### ISSUE TITLE
Error whilst editing an html import

### ISSUE THREAD

**Issue by akc42** [Polymer/vscode-plugin#33]
I was editing app/src/pas-appointments.html, trying to add a new html import via a snippet. I mispelled paper-scroll-header-panel, but I am not sure quite what.  Whilst I was doing this the output window popped up with multiple (approx 15) entries like to following in it.

```
[Error - 08:16:34] Error: Internal error: document app/src/pas-appointments.html already exists
    at Analyzer._makeDocument (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/analyzer.js:147:19)
    at Analyzer.<anonymous> (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/analyzer.js:132:39)
    at next (<anonymous>)
    at fulfilled (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/analyzer.js:17:58)
```

**Comment by akc42** [comment]
It happened again after I completely screwed up editing the file, refreshed in from git HEAD and then started to add this html import again.  This time I didn't make an error typing - rather I paused whilst typing out paper-scroll-head... (with two cursors) and at that point it happened again.  As I started typing again ...er, each character that I typed caused a new error message.

I tried to replicate this by removing the line and starting again, but it wouldn't repeat.

I then closed down vscode and started again and sure enough it came back

One other possibly salient fact is that I have autosave on - which may mean the trigger is related to saving the file after one of the pauses.

**Comment by akc42** [comment]
I have traced a few problems to having autoSave switched on with `afterDelay`.  I have now changed that to `onFocusChange`.  Nevertheless, today the same issue seems to arise when backspacing in a comment.

```
/**
  * This is an array of weeks in the displayed month.  It is by manipulating this array
  * that we show the month.  There can be up to 6 weeks in a month because we always start_page_appo
   */
```
I had just typed the "start" and I think I then hit enter to start a new line.  Instead the editor completed the text with what you see and I was backspacing to get rid of it.  I have the Document This extension installed which is why the comment was fully completed, as it does that when you type the initial "/**"

**Comment by rictic** [comment]
We don't autocompeter inside JavaScript currently, this is most likely a different plugin, or vscode's built in word based (nonsemantic) autocompeter.

**Comment by akc42** [comment]
It not the autocomplete that is the problem per se, its the backspacing causing the error to come up.

**Comment by rictic** [comment]
Ah, misread while foggy-headed in the morning! Sorry about that.

**Comment by akc42** [comment]
This is a similar error, but this time I am in a file with lots of lazy-imports links at the head
PAS.LazyPage is a behavior that I import into any element which has an iron-pages as one of its children and a selection of elements underneath.  All of them are loaded using the importLazyGroup mechanism and selected from an `<akc-route>` element (similar in concept to `<app-route>`.

This error is occuring as I mouse over one of the sub-elements of `iron-pages`
 
```
[Error - 21:34:44] Error: Expected to find at most one behavior with id PAS.LazyPage but found 3.
    at InlineDocument.getOnlyAtId (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/model/document.js:156:19)
    at _getFlattenedAndResolvedBehaviors (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/polymer/polymer-element.js:123:40)
    at getFlattenedAndResolvedBehaviors (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/polymer/polymer-element.js:118:5)
    at resolveElement (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/polymer/polymer-element.js:105:34)
    at ScannedPolymerElement.resolve (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/polymer/polymer-element.js:67:16)
    at InlineDocument.resolve (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/model/document.js:102:48)
    at ScannedInlineDocument.resolve (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/model/inline-document.js:41:24)
    at Document.resolve (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/model/document.js:102:48)
    at Analyzer._makeDocument (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/analyzer.js:154:18)
    at Analyzer.<anonymous> (/home/alan/.vscode/extensions/polymer.polymer-ide-0.1.2/server/node_modules/polymer-analyzer/lib/analyzer.js:132:39)
```

**Comment by TimvdLippe** [comment]
This error is also reported when using the `atom-plugin`, so I think this is an issue with `polymer-editor-service` instead.

**Comment by rictic** [comment]
I believe this is fixed as of v0.3.1, please open a new issue if you see it again!
